medusa/search/proper.py

```diff
@@ -159,6 +159,14 @@ def _get_proper_results(self):  # pylint: disable=too-many-locals, too-many-bran
                 log.info('Skipping non-proper: {name}', {'name': cur_proper.name})
                 continue
 
+            if not cur_proper.show.episodes.get(cur_proper.parse_result.season_number) or \
+                    any([ep for ep in cur_proper.parse_result.episode_numbers
+                         if not cur_proper.show.episodes[cur_proper.parse_result.season_number].get(ep)]):
+                log.info('Skipping proper for wrong season/episode: {name}', {'name': cur_proper.name})
+                if cur_proper.name not in processed_propers_names:
+                    self.processed_propers.append({'name': cur_proper.name, 'date': cur_proper.date})
```

Comment:
I don't think you can do this here. What if it finds a proper for another episode that is legitimately needed (eg. within the proper range)? It shouldn't be ignored for every subsequent proper search.

Reply:
@medariox maybe we shouldn't add this to self.processed_propers so it would work as you said. What do you think?

Reply:
You can't just skip the proper result here? Isn't that what we already did?

Reply:
What medariox said it in that loop im looking for S01E02 proper. So if is a season/episode mismatch like S01E01, it will be skipped. But in the next item, we are searching for S01E01, it will never get that proper because its on the "already processed propers"

Reply:
aah I understand, and yes I agree. We should ignore it, but not throw it away.